src/lib/dnssd/Resolver.h

```diff
@@ -295,6 +303,39 @@ enum class DiscoveryType
     kCommissionerNode
 };
 
+/**
+ * A delegate that can be notified of node additions/removals as a mdns browse proceeds.
+ */
+class BrowseDelegate
+{
+public:
+    virtual ~BrowseDelegate() {}
+
+    /**
+     * @brief
+     *   Called when a node is added.
```

Comment:
please cleanup doccomments to help reading code instead of saying the obvious (otherwise it takes up vertical space, which is worse for readability). it is clear that `nodeData is "the node data"` also it is clear that `OnBrowseAdd` is called when a node data is added during a browse operation.